pkg/apis/management.cattle.io/v3/authn_types.go

```diff
@@ -29,22 +29,22 @@ const (
 type Token struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Token           string            `json:"token" norman:"writeOnly,noupdate"`
-	UserPrincipal   Principal         `json:"userPrincipal" norman:"type=reference[principal]"`
-	GroupPrincipals []Principal       `json:"groupPrincipals,omitempty" norman:"type=array[reference[principal]]"`
-	ProviderInfo    map[string]string `json:"providerInfo,omitempty"`
-	UserID          string            `json:"userId" norman:"type=reference[user]"`
-	AuthProvider    string            `json:"authProvider"`
-	TTLMillis       int64             `json:"ttl"`
-	LastUsedAt      *metav1.Time      `json:"lastUsedAt,omitempty"`
-	IsDerived       bool              `json:"isDerived"`
-	Description     string            `json:"description"`
-	Expired         bool              `json:"expired"`
-	ExpiresAt       string            `json:"expiresAt"`
-	Current         bool              `json:"current"`
-	ClusterName     string            `json:"clusterName,omitempty" norman:"noupdate,type=reference[cluster]"`
-	Enabled         *bool             `json:"enabled,omitempty" norman:"default=true"`
+	Token             string            `json:"token" norman:"writeOnly,noupdate"`
+	UserPrincipal     Principal         `json:"userPrincipal" norman:"type=reference[principal]"`
+	GroupPrincipals   []Principal       `json:"groupPrincipals,omitempty" norman:"type=array[reference[principal]]"`
+	ProviderInfo      map[string]string `json:"providerInfo,omitempty"`
+	UserID            string            `json:"userId" norman:"type=reference[user]"`
+	AuthProvider      string            `json:"authProvider"`
+	TTLMillis         int64             `json:"ttl"`
+	LastUsedAt        *metav1.Time      `json:"lastUsedAt,omitempty"`
+	LastIdleTimeout   metav1.Time       `json:"lastIdleTimeout,omitempty"`
```

Comment:
As with UserActivity fields the field doesn't read or covey the message well. I'd recommend something like `ActivityLastSeenAt`. But most importantly it seems like it has to be a reference type `*metav1.Time` as it makes sense *only* for session tokens and all existing tokens don't have this field.